src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -319,6 +318,16 @@ void AppleseedInteractiveRender::BeginSession()
     else
         get_view_params_from_viewport(view_params, *GetViewExp(), m_time);
 
+    RendererSettings renderer_settings = RendererSettings::defaults();
+    
+    Renderer* curr_renderer = GetCOREInterface()->GetCurrentRenderer(false);
+    if (curr_renderer && curr_renderer->ClassID() == AppleseedRenderer::get_class_id())
```

Comment:
Probably cannot happen. Should I remove whole if statement or should still check `if (curr_renderer)`?

Reply:
If it cannot happen for sure, you should remove the entire statement.

In fact, having a closer look at it, the entire statement is a no-op since `AppleseedRenderer::get_renderer_settings()` returns a _copy_ of the renderer settings, which you then modify, which then get lost. So the whole thing is doing nothing :)

Reply:
Hi @dictoon, I only modify one setting after I get renderer_settings from main renderer, all others are kept aren't they? I certainly see them carried from main renderer when I change settings in the dialog.
Function declared like this
`const RendererSettings& AppleseedRenderer::get_renderer_settings()` It returns a reference I think.. But the assignment makes a copy, is that what you mean?
Should I do 
`RendererSettings& renderer_settings = appleseed_renderer->get_renderer_settings();` instead?

Reply:
The way you ended up doing it is correct:

- `AppleseedRenderer::get_renderer_settings()` returns a constant reference to the settings, i.e. it gives _read only_ access to the settings (which is the right thing to do, we don't want users of this class to be able to modify the renderer settings).

- When you do

      RendererSettings renderer_settings = AppleseedRenderer::get_renderer_settings();

  the settings are indeed copied.

- This wouldn't compile at all:

      RendererSettings& renderer_settings = AppleseedRenderer::get_renderer_settings();

  You can't convert a constant reference to a mutable (non-constant) one.